spread.yaml

```diff
@@ -339,13 +345,13 @@ prepare: |
 
     # NOTE: At this stage the source tree is available and no more special
     # considerations apply.
-    "$TESTSLIB"/prepare-restore.sh --prepare-project
+    ( . "$TESTSLIB"/prepare-restore.sh --prepare-project )
```

Comment:
why you are changing this?

Reply:
We need to do a system upgrade in Arch in order to get a consistent system state. I figured that moving this operation into project prepare was the best fit (see https://github.com/snapcore/snapd/pull/4285/commits/77730506789585c4c253147278db63df106196d0 for implementation). However, there is a slight complication, in order to do a reboot properly, we use the `REBOOT` command which is only defined at the top level spread `prepare|restore..` and so on, so it's not visible in `prepare-restore.sh` which is executed in a separate shell. To work around this problem, I've opted for sourcing the script instead. 😞 

Other options are to move the upgrade piece to `prepare` block, or use some special exit code in `prepare-restore.sh` to indicate that reboot is needed.

Reply:
Could you please add a comment (copy/paste this here is probably fine) about the details why this is done in the way you do it?